Use current database in MySQL's `table_exists?`

mikecmpbll opened the pull request:
This is a minor regression from when we switched from using SHOW TABLES
to querying the information_schema for `table_exists?`. SHOW TABLES by
default querys the currently selected database but the current
implementation defaults to `@config[:database]` if no database name is
prepended to the table name. This meant that if you issued a `USE`
statement to change database, `table_exists?` gives spurious results as
it's still querying `@config[:database]`.

The solution is to use the MySQL `DATABASE()` function in place of
`@config[:database]`.

https://github.com/rails/rails/pull/18856#issuecomment-286064885

kamipo commented:
FYI: #28068 also fixes this issue.
https://github.com/rails/rails/pull/28068/files#diff-8b6f13a2f84b84c914f735315609ce24R19

mikecmpbll commented:
@kamipo derp; didn't spot that one. shall i close?

rafaelfranca commented:
This change is preferable to 5-0-stable. Could you open it against that branch?

Commit:
Use current database in MySQL's `table_exists?`

This is a minor regression from when we switched from using SHOW TABLES
to querying the information_schema for `table_exists?`. SHOW TABLES by
default querys the currently selected database but the current
implementation defaults to `@config[:database]` if no database name is
prepended to the table name. This meant that if you issued a `USE`
statement to change database, `table_exists?` gives spurious results as
it's still querying `@config[:database]`.

The solution is to use the MySQL `DATABASE()` function in place of
`@config[:database]`.

mikecmpbll commented:
@rafaelfranca done 🐎
